src/components/ont/runs/ONTFlowcell.vue

```diff
@@ -85,110 +83,73 @@ export default {
       required: true,
     },
   },
-  data() {
-    return {
-      /**Represents the current flowCell Validation state */
-      flowCellValidationState: {
-        statusId: FieldStatusEnum.Empty,
-        statusBarcode: FieldStatusEnum.Empty,
-        errorBarcode: '',
-        errorId: '',
-        state: FlowCellStateEnum.Empty,
-      },
-    }
-  },
   computed: {
     ...mapState(useOntRunsStore, ['currentRun', 'getFlowCell']),
     // For Vuex asynchronous validation we need to use computed getter and setter properties
     barcode: {
       get() {
-        return this.poolTubeBarcode
+        return this.flowCell.tube_barcode
       },
       async set(value) {
-        /*It is required to update the barcode here because components are externally
-          listening to this state */
-        this.setPoolTubeBarcode({
-          barcode: value,
-          position: this.position,
-        })
-        //Validate barcode field of flowcell
-        const validation = await this.flowCell.validateFlowCell(
-          false,
-          true,
-          this.flowCellValidationState,
-        )
-        this.setFlowCellValidation(validation)
+        this.flowCell.tube_barcode = value
+        await this.fetchPool(value)
+        this.flowCell.validateBarcode()
       },
     },
     flowcellId: {
       get() {
-        const flowCell = this.getFlowCell(this.position)
-        if (flowCell) {
-          return this.getFlowCell(this.position).flowcell_id
-        }
-        return ''
+        return this.flowCell.flowcell_id
       },
       async set(value) {
-        this.setFlowcellId({ $event: value, position: this.position })
-        //Validate flowcell id field of flowcell
-        const validation = await this.flowCell.validateFlowCell(
-          true,
-          false,
-          this.flowCellValidationState,
-        )
-        this.setFlowCellValidation(validation)
+        this.flowCell.flowcell_id = value
+        this.flowCell.validateFlowCellId()
       },
     },
     flowCell() {
       //If flowcell is not present in the store, return a new empty flowcell
       //(this is reduce the number of null checks in validations)
-      return this.getFlowCell(this.position) ?? { ...flowCellType }
-    },
-
-    poolTubeBarcode() {
-      return this.flowCell.tube_barcode
-    },
-    flowcellIdValidationError() {
-      return this.flowCellValidationState.errorId
-    },
-    isFlowcellIdExists() {
-      return this.flowCellValidationState.statusId !== FieldStatusEnum.Empty
-    },
-    barcodeValidationError() {
-      return this.flowCellValidationState.errorBarcode
-    },
-    isBarcodeExists() {
-      return this.flowCellValidationState.statusBarcode !== FieldStatusEnum.Empty
+      return this.getFlowCell(this.position) ?? { ...flowCellType() }
     },
     /**Displays green if valid, red if invalid and no border if empty */
     flowcell_id_field_colour() {
```

Comment:
I was sure Vue had a really clever way of managing dynamic classes but not sure how it would work here.

Reply:
Some interesting stuff in https://vuejs.org/guide/essentials/class-and-style. Not sure how much of an improvement it is though.